primitives/contract/src/mapping_token_factory/s2s.rs

```diff
@@ -67,7 +68,7 @@ impl Sub2SubMappingTokenFactory {
 /// @original_token: the origin token address
 /// @recipient: the final receiver of the token to be unlocked on the source chain
 /// @amount: the amount of the unlocked token
-#[derive(Debug, PartialEq, Eq)]
+#[derive(Encode, Decode, Debug, PartialEq, Eq, Clone)]
```

Comment:
why do we need to add this Encode and Decode feature?

Reply:
O, We don't need this after changing the recipient type to Vec. Great catch.

Reply:
Done